src/modules/ens/ens.service.ts

```diff
@@ -159,17 +159,17 @@ export class EnsService {
     owner?: string;
   }) {
     try {
-
+      if (!owner  || owner !== emptyAddress) {// prevents attempt to resync after namepsace deregisteration
```

Comment:
The reason for allowing synchronisation when there is no owner is that the `owner` param is optional(like you mentioned the presence of the owner param depends on the event ) and so makes sense for it to try to resolve that below [here](https://github.com/energywebfoundation/iam-cache-server/blob/1a344c61e64b606d255cf935cd0021112b137b48/src/modules/ens/ens.service.ts#L257).

What namespace should be removed? I think there is a function to handle that already  removal of namespace from the db and this method isn't responsible for that so I'm not sure what duplication you are referring to.

This only ensures that namespaces that have been removed do not attempt to resync again because it throws an error.